File: Adapter/AdapterPatternRectangle.py
from abc import ABC, abstractmethod

class Shape(ABC):
    @abstractmethod
    def draw(self):
        pass

class Circle(Shape):
    def __init__(self, x, y, r):
        self.x = x
        self.y = y
        self.radious = r

    def draw(self):
        print("Draw the Circle.")


class Rectange:
    def __init__(self, x, y, l, w):
        self.x = x
        self.y = y
        self.length = l
        self.width = w

    def oldDraw(self):
        print("Drawing Rectangle.")
        

class RectangeAdapter(Shape):
    def __init__(self, x, y, l, w):
        self._adaptee = Rectange(x, y, l, w)
        
    def draw(self):
        self._adaptee.oldDraw()


# Client Code
adapter = RectangeAdapter(1,2,3,4)
adapter.draw()


"""
Output:
Drawing Rectangle.

"""

File: Adapter/AdapterPatternRectangle.java
// Desired Interface
interface Shape {
    void draw();
}

// Circle class
class Circle implements Shape {
    private int x, y, radius;

    public Circle(int x, int y, int r) {
        this.x = x;
        this.y = y;
        this.radius = r;
    }

    @Override
    public void draw() {
        System.out.println("Draw the Circle.");
    }
}

// Rectangle class (Adaptee)
class Rectangle {
    private int x, y, length, width;

    public Rectangle(int x, int y, int l, int w) {
        this.x = x;
        this.y = y;
        this.length = l;
        this.width = w;
    }

    public void oldDraw() {
        System.out.println("Drawing Rectangle.");
    }
}

// RectangleAdapter class
class RectangleAdapter implements Shape {
    private Rectangle adaptee;

    public RectangleAdapter(int x, int y, int l, int w) {
        this.adaptee = new Rectangle(x, y, l, w);
    }

    @Override
    public void draw() {
        adaptee.oldDraw();
    }
}

// Client Code
public class AdapterPatternRectangle {
    public static void main(String[] args) {
        Shape adapter = new RectangleAdapter(1, 2, 3, 4);
        adapter.draw();
    }
}
